src/searchbar/SearchBar-ios.js

```diff
@@ -48,28 +56,26 @@ class SearchBar extends Component {
   onFocus = () => {
     this.props.onFocus();
     UIManager.configureNextLayoutAnimation && LayoutAnimation.easeInEaseOut();
-    this.setState({ hasFocus: true });
+    this.setState({
+      hasFocus: true,
+      cancelButtonTransform: -this.state.cancelButtonWidth,
+    });
   };
 
   onBlur = () => {
     this.props.onBlur();
     UIManager.configureNextLayoutAnimation && LayoutAnimation.easeInEaseOut();
-    this.setState({ hasFocus: false });
+    this.setState({
+      hasFocus: false,
+      cancelButtonTransform: this.state.cancelButtonWidth,
```

Comment:
Just curious to know why is this initialised to 0 and set to `cancelButtonWidth` on blur ? Shouldn't we set it to 0 on blur or initialise it to `cancelButtonWidth` with the `onLayout` ?

Reply:
It's using marginLeft for the button 
https://github.com/react-native-training/react-native-elements/pull/1218/files#diff-348f551797bae8f23921aab3699399eeR145

So on blur it needs to move back -> way to be hidden. So that's why it uses the buttonWidth and not 0

Reply:
Yea but I mean, the "blur" state is somehow the default state right, as it is not focused ?

So at the beginning the `marginLeft` will be 0 -> you focus -> you blur -> it will now be `cancelButtonWidth`. Is it the expected behaviour ?

Reply:
I think I understand what you're saying. Curious doing it that way would cause an additional animation on mount, since the `cancelButtonTransform` would then be set to a value after the onlayout. Think that may have a blip in the UI.

Reply:
Yea but you could have put 0 here to have the same "slide" effect for all the focuses (the first and the followings). Here you have a more important "slide effect" for the next focus cause the `Button` is coming from further than on the first focus ([repro snack](https://snack.expo.io/HktaWifyQ))